problem_builder/step.py

```diff
@@ -75,40 +83,109 @@ class StepMixin(object):
         scope=Scope.content
     )
 
+    editable_fields = ('display_name', 'show_title',)
+
+    def _(self, text):
+        """ Translate text """
+        return self.runtime.service(self, "i18n").ugettext(text)
+
     @lazy
-    def step_number(self):
-        return list(self.get_parent().steps).index(_normalize_id(self.scope_ids.usage_id)) + 1
+    def siblings(self):
+        return self.get_parent().steps
 
     @lazy
-    def lonely_step(self):
-        if _normalize_id(self.scope_ids.usage_id) not in self.get_parent().steps:
-            raise ValueError("Step's parent should contain Step", self, self.get_parent().steps)
-        return len(self.get_parent().steps) == 1
+    def child_number(self):
+        return list(self.siblings).index(_normalize_id(self.scope_ids.usage_id)) + 1
+
+    @lazy
+    def lonely_child(self):
+        if _normalize_id(self.scope_ids.usage_id) not in self.siblings:
+            message = u"{child_caption}'s parent should contain {child_caption}".format(child_caption=self.CAPTION)
+            raise ValueError(message, self, self.siblings)
+        return len(self.siblings) == 1
```

Comment:
@itsjeyd child_number and lonely_child are quite similar to what StepMixix provide - I remember extracting these into a common mixin. I believe it would make sense to extract it.